datasource/vmware/vmware_test.go

```diff
@@ -165,10 +182,36 @@ func TestFetchUserdata(t *testing.T) {
 			},
 			err: errors.New(`Unsupported encoding "test encoding"`),
 		},
+		{
+			variables: map[string]string{
+				"coreos.config.url": ts.URL + "/good",
+			},
+			userdata: "test config",
+		},
+		{
+			variables: map[string]string{
+				"coreos.config.url": ts.URL + "/bad",
+			},
+			err: pkg.ErrNotFound{fmt.Errorf("Not found. HTTP status code: %d", http.StatusNotFound)},
+		},
+		{
+			variables: map[string]string{
+				"coreos.config.url": ts.URL + "/ugly",
+			},
+			err: pkg.ErrServer{fmt.Errorf("Server error. HTTP status code: %d", http.StatusInternalServerError)},
+		},
+	}
+
+	var downloader urlDownloadFunction = func(url string) ([]byte, error) {
```

Comment:
Yes. I think this should just test that the data and errors are correct. You could then test `urlDownload` separately.